src/appsettings/settings.py

```diff
@@ -470,6 +472,8 @@ def raw_value(self):
             AttributeError: if the variable is missing.
             KeyError: if the item is missing from nested setting.
         """
+        if self.full_name in os.environ:
+            return self.decode_environ(os.environ[self.full_name])
         if isinstance(self.parent_setting, NestedDictSetting):
             return self.parent_setting.raw_value[self.full_name]
```

Comment:
`parent_setting` needs to take precedence over environment, so the subsettings are not incorrectly read from unrelated environment values. Test for that is needed.

Reply:
> Subsettings should not load themselves from environment. We probably missed that in design.

Done. Test for it has been written.

Reply:
If the conditions are switched, the `parent_setting` doesn't have to be checked twice. Also it will be more legible.

Reply:
Done.